flexget/plugins/clients/tarantool_queue.py

```diff
@@ -0,0 +1,119 @@
+from loguru import logger
+from flexget.entry import Entry
+from flexget.event import event
+from flexget import plugin
+
+logger = logger.bind(name='tarantool_queue')
+
+class TarantoolTube:
+    def __init__(self, queue, tube_name):
+        self._queue = queue
+        self._tube_name = tube_name
+
+    def put(self, data):
+        return self._queue.call('queue.tube.{0}:put'.format(self._tube_name), data)
+
+    def take(self, timeout=None):
+        entries = self._queue.call('queue.tube.{0}:take'.format(self._tube_name), timeout)
+        if entries is None:
+            entries = []
+        return entries
+
+    def ack(self, task_id):
+        return self._queue.call('queue.tube.{0}:ack'.format(self._tube_name), task_id)
+
+class TarantoolQueue:
+    def __init__(self, tarantool_module, host, port, user, password):
+        self._server = tarantool_module.connect(host, port, user=user, password=password)
+        self._tubes = {}
+
+    def get_tube(self, tube_name):
+        if tube_name in self._tubes:
+            tube = self._tubes[tube_name]
+        else:
+            tube = TarantoolTube(self, tube_name)
+            self._tubes[tube_name] = tube
+        return tube
+
+    def call(self, function, *args):
+        return self._server.call(function, *args)
+
+class PluginTarantoolQueueBase:
+    schema = {
+        'anyOf': [
+            {'type': 'boolean'},
+            {
+                'type': 'object',
+                'properties': {
+                    'host': {'type': 'string'},
+                    'port': {'type': 'integer'},
+                    'user': {'type': 'string'},
+                    'password': {'type': 'string'},
+                    'tube': {'type': 'string'},
+                    'chunk_size': {'type': 'integer', 'default': 30}
```

Comment:
You'll want to setup the schema with `required` fields. It looks like, as of now, the config would pass even with missing fields.  More info can be found here (https://flexget.readthedocs.io/en/latest/develop/schemas.html) or message me on slack and I can assist.